lib/gitsh/cli.rb

```diff
@@ -82,5 +84,15 @@ def option_parser
         end
       end
     end
+
+    def ensure_executable_git
+      IO.popen(env.git_command).close
+    rescue Errno::ENOENT
```

Comment:
We should possibly also rescue from `Errno::EACCES` here, with a different error message indicating the supposed Git binary wasn't executable.